backend/layers/persistence/persistence.py

```diff
@@ -344,30 +362,41 @@ def get_all_collections_versions(self, get_tombstoned: bool = False) -> Iterable
                 )
 
             result = []
+            all_dataset_tombstones = {
+                str(dataset.id)
+                for dataset in session.query(DatasetTable).filter(DatasetTable.tombstone.is_(True)).all()
+            }
+            all_dataset_version_mappings = {
+                str(dataset_version.id): str(dataset_version.dataset_id)
+                for dataset_version in session.query(DatasetVersionTable).all()
```

Comment:
Ok I think the point about doing the dataset tombstone filtering elsewhere is a good one because, at present, this function is only being used to generate the Collections index, so we can wait until only mapped versions have been selected, and then we can pare it down.

Reply:
Ok the issue here is that the business layer functions that are calling this should be using separate functions, independent of my tombstoning changes, though I agree my tombstoning changes make things a little worse